distribution/docker/Dockerfile

```diff
@@ -17,7 +17,7 @@
 # under the License.
 #
 
-ARG JDK_VERSION=11
+ARG JDK_VERSION=17
 
 # The platform is explicitly specified as x64 to build the Druid distribution.
```

Comment:
Thank you @georgew5656! I was having real issues getting this to work on my M2 ;)

Is `--platform` still required here?

Reply:
i don't think we actually support multiarch images right now, when I go to the druid dockerhub i only see a amd64 image which would explain why it doesn't work on a arm based mac

Reply:
@abhishekagarwal87 do you know how this image gets built? is it a pipeline somewhere?

Reply:
it's a bit of black box for me as well. Apache infra jobs builds and pushes the image as far as I know.

Reply:
What I shoulda said (!) was I managed to build an arm Docker image using your fix - but yeah the official image would be nicer :)

Reply:
Thanks @abhishekagarwal87

Reply:
I think we can raise an Infra ticket, or ask in ASF Slack, to adjust how the images get built. (Or at least ask how they get built.) If it's possible to publish ARM images too that would be nice.

Reply:
e.g. what this repo is showing https://github.com/cgiraldo/docker-hello-multiarch

Reply:
Another option might be to move the docker image building to a github action on tags only, have some action that calls the appropriate `buildx` command, and pushes a multi-arch image to dockerhub

Reply:
The automated dockerhub builds seem nice because then we don't have to worry about dockerhub credentials in our GHA stuff.